pkg/logs/read.go

```diff
@@ -0,0 +1,103 @@
+package logs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/observatorium/up/pkg/auth"
+	"github.com/observatorium/up/pkg/options"
+	"github.com/observatorium/up/pkg/transport"
+	"github.com/observatorium/up/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+func Read(
+	ctx context.Context,
+	endpoint *url.URL,
+	tp auth.TokenProvider,
+	labels []prompb.Label, // change to Loki ProtoBufs
```

Comment:
Is this a todo waiting for a compatible prometheus version to be merged into loki?

Reply:
yes exactly.